platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,118 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[no_source_in_root]]
+== Do Not Put Source Files in the Root Project
```

Comment:
I feel like this is not just about sources, in general, you should avoid applying any plugins to the root project, including Java plugin. Should we broaden to that?